pages/api/accounts/[id]/index.js

```diff
@@ -53,11 +53,15 @@ export default async (req, res) => {
   }
 
   if (req.method === 'DELETE') {
+    if (id === userId) {
```

Comment:
The request passes through because of [IsStrictlyEqual](https://tc39.es/ecma262/#sec-isstrictlyequal)
![image](https://user-images.githubusercontent.com/21258296/204342306-fdbfc60a-3022-4cf1-826c-24cc953a882d.png)